### PR TITLE
fix broken links in operator doc.

### LINKED ISSUE #27787: Update istio/istio operator README to the latest
- [ ] `Component configuration API` points to this invalid URL: https://github.com/istio/api/blob/master/operator/v1alpha1/component.proto

- [ ] legacy `Helm installation API` points to this URL (which is also 404 page): https://istio.io/docs/reference/config/installation-options/

- [ ] no `operator/deploy/` repo on master

### PR DESCRIPTION
**Pull request by morvencao**
resolve: https://github.com/istio/istio/issues/27787

[ ] Configuration Infrastructure
[ X ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by carolynhu — COMMENTED** [review]
@morvencao Thanks for this fix! It helps the new operator developers a lot. LGTM!

File: operator/README.md [hunk 1]
```diff
@@ -20,12 +20,12 @@ three main components:
 
 - [MeshConfig](https://github.com/istio/api/blob/master/mesh/v1alpha1/config.proto) for runtime config consumed directly by Istio
 control plane components.
-- [Component configuration API](https://github.com/istio/api/blob/master/operator/v1alpha1/component.proto), for managing
+- [Component configuration API](https://github.com/istio/api/blob/00671adacbea20f941cb20cce021bc63cbad1840/operator/v1alpha1/operator.proto#L137-L149), for managing
```
Comment (shamsher31) [inline operator/README.md:23]: Should it point to
https://github.com/istio/api/blob/00671adacbea20f941cb20cce021bc63cbad1840/operator/v1alpha1/operator.proto#L42-L93 ?

Comment (morvencao) [reply operator/README.md]: Thanks, updated.

**Review by shamsher31 — APPROVED** [review]
Thanks

**Review by richardwxn — APPROVED** [review]

**Pull request merged**